controllers/etcdbackupschedule_controller.go

```diff
@@ -2,32 +2,136 @@ package controllers
 
 import (
 	"context"
+	"time"
 
 	"github.com/go-logr/logr"
+	"github.com/robfig/cron/v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	etcdv1alpha1 "github.com/improbable-eng/etcd-cluster-operator/api/v1alpha1"
 )
 
+type CronScheduler interface {
+	AddFunc(spec string, cmd func()) (cron.EntryID, error)
+	Remove(id cron.EntryID)
+}
+
 // EtcdBackupScheduleReconciler reconciles a EtcdBackupSchedule object
 type EtcdBackupScheduleReconciler struct {
 	client.Client
 	Log logr.Logger
+
+	// CronHandler is able to schedule cronjobs to occur at given times.
+	CronHandler CronScheduler
+
+	// Schedules holds a mapping of resources to the object responsible for scheduling the backup to be taken.
+	Schedules map[string]Schedule
```

Comment:
Does this need to be exported? I see numerous operations on the map, which become race-y if anything else interacts with the map concurrently

Reply:
Hm, *additionally*, we could write to this map in multiple goroutines as Reconcile can be called concurrently. I think we need to gate writes to the map with a `sync.RWLock` or something 😅

Reply:
Do we run our tests with the race detector enabled...?

Reply:
yep you're right. ive swapped this out for a thread safe wrapper. we don't run the race detector (doing so showed ~4 potential races, i'll make a ticket to address).